From 79495daac64362202abce00379f899069d1ca1a5 Mon Sep 17 00:00:00 2001
From: Martin Gerhardy <martin.gerhardy@gmail.com>
Date: Sun, 8 Jun 2014 17:21:58 +0200
Subject: [PATCH 07/40] * fixed narrowing compile error with clang

* case value evaluates to -1, which cannot be narrowed to type 'GLenum' (aka 'unsigned int')
---
 src/client/renderer/r_state.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/client/renderer/r_state.cpp b/src/client/renderer/r_state.cpp
index 71d966f542..d03a52cf11 100644
--- a/src/client/renderer/r_state.cpp
+++ b/src/client/renderer/r_state.cpp
@@ -147,7 +147,8 @@ void R_UseMaterial (const material_t* material)
 
 void R_BindArray (GLenum target, GLenum type, const void* array)
 {
-	switch (target) {
+	const int v = static_cast<int>(target);
+	switch (v) {
 	case GL_VERTEX_ARRAY:
 		glVertexPointer(COMPONENTS_VERTEX_ARRAY3D, type, 0, array);
 		break;
@@ -180,7 +181,8 @@ void R_BindArray (GLenum target, GLenum type, const void* array)
  */
 void R_BindDefaultArray (GLenum target)
 {
-	switch (target) {
+	const int v = static_cast<int>(target);
+	switch (v) {
 	case GL_VERTEX_ARRAY:
 		R_BindArray(target, GL_FLOAT, r_state.vertex_array_3d);
 		break;
-- 
2.16.1

